ruby/ext/google/protobuf_c/message.c

```diff
@@ -157,6 +185,34 @@ static int extract_method_call(VALUE method_name, MessageHeader* self,
   bool has_field = upb_msgdef_lookupname(self->descriptor->msgdef, name, name_len,
 			                                   &test_f, &test_o);
 
+  // Look for wrapper type accessor of the form <field_name>_as_value
+  if (!has_field &&
+      (accessor_type == METHOD_GETTER || accessor_type == METHOD_SETTER) &&
+      name_len > 9 && strncmp(name + name_len - 9, "_as_value", 9) == 0) {
+    // Find the field name
+    char wrapper_field_name[name_len - 8];
+    strncpy(wrapper_field_name, name, name_len - 9);
+    wrapper_field_name[name_len - 7] = '\0';
```

Comment:
`wrapper_field_name` is only `name_len - 8` wide.  `name_len - 7` is greater than `name_len - 8`, so I think this is writing outside the buffer.